torch/csrc/jit/tensorexpr/cuda_codegen.h

```diff
@@ -34,42 +38,47 @@ class CudaAnalysis : public IRVisitor {
     return cross_block_bufs_;
   }
 
+  const std::vector<const Expr*>& gpu_block_extents() const {
+    return gpu_block_extents_;
+  }
+
+  const std::vector<const Expr*>& gpu_thread_extents() const {
+    return gpu_thread_extents_;
+  }
+
  private:
   void visit(const Store* v) override {
     store_targets_.insert(v->buf());
   }
 
   void visit(const Allocate* v) override;
   void visit(const Free* v) override;
+  void visit(const For* v) override;
 
   std::unordered_set<const Buf*> store_targets_;
   std::unordered_set<const Var*> thread_local_bufs_;
   std::unordered_set<const Var*> cross_block_bufs_;
+
+  std::vector<const Expr*> gpu_block_extents_;
+  std::vector<const Expr*> gpu_thread_extents_;
 };
 
 // An IRMutator that replaces binding loop options with Cuda metavars.
 class GPUMetaVarRewriter : public IRMutator {
  public:
-  explicit GPUMetaVarRewriter() {
+  explicit GPUMetaVarRewriter(const CudaAnalysis* cuda_analysis)
+      : cuda_analysis_(cuda_analysis) {
     gpu_block_vars_ = {new Var("blockIdx.x", kInt),
                        new Var("blockIdx.y", kInt),
                        new Var("blockIdx.z", kInt)};
     gpu_thread_vars_ = {new Var("threadIdx.x", kInt),
-                        new Var("threadIdx.y", kInt),
+                        new Var("threadidx.y", kInt),
```

Comment:
Lowercase `i` in threadidx.y looks like a typo

Reply:
I need to rebind `u` in visual mode, easy to screw up case. Good catch.